# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-xcb-connection-shm.c
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-xcb-connection-shm.c
--- a/gfx/cairo/cairo/src/cairo-xcb-connection-shm.c	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-xcb-connection-shm.c	Mon May 06 19:25:12 2024 +0000
@@ -43,7 +43,7 @@
 				  uint32_t id,
 				  cairo_bool_t readonly)
 {
-    uint32_t segment = _cairo_xcb_connection_get_xid (connection);
+    uint32_t segment = xcb_generate_id (connection->xcb_connection);
     assert (connection->flags & CAIRO_XCB_HAS_SHM);
     xcb_shm_attach (connection->xcb_connection, segment, id, readonly);
     return segment;
@@ -109,7 +109,6 @@
 {
     assert (connection->flags & CAIRO_XCB_HAS_SHM);
     xcb_shm_detach (connection->xcb_connection, segment);
-    _cairo_xcb_connection_put_xid (connection, segment);
 }
 
 #endif /* CAIRO_HAS_XCB_SHM_FUNCTIONS */